### ISSUE TITLE
Improve predictability of `pack:` names

### ISSUE THREAD

**Issue by muchzill4** [Open-CMSIS-Pack/devtools#941]
## Problem Statement
The csolution files can specify `pack:` entries in various of ways:

1. Predictable version
```yaml
- pack: Vendor::Name@1.2.3 # specific version of a pack
- pack: Vendor::Name # latest version of a pack
- pack: Vendor::Name@>=1.2.3 # latest version of a pack, but greater or equal to 1.2.3
```

2. Unpredictable version
```yaml
- pack: Vendor # all packs of the "Vendor" are considered, version is non deterministic
- pack: Vendor::Nam* # all packs of the "Vendor" that match "Nam*" wildcard are considered, version is non deterministic
```

3. No predictability at all
`packs:` section is not explicitly specified - entire pack repository available to the tool is considered. This means packs from all vendors with arbitrary versions.

The universe of packs is big. The bigger the repository available to the user and the tooling they use, the harder it becomes to predict what version of a pack will be used. We already have use-cases, where all packs in the universe are available. We haven't seen the unpredictable `pack:` names used in the wild.

I believe that being able to reason about the pack versions used in a clear and concise way would be valuable. It is also a great opportunity to simplify and tighten the specification before the 2.0 release.

## Proposal 1
Remove support for the least predictable specifiers of `pack:` entries. In other words, disallow:
```yaml
- pack: Vendor # vendor only
- pack: Vendor::Nam* # any wildcards
```

I’d also suggest removing the ability to not specify `packs:` at all, i.e.: equivalent of doing:
```yaml
- pack: *
```

## Proposal 2
A slightly less drastic approach would be to always consider latest version available of a pack that matches the specifier, unless a version is explicitly specified. This means:
```yaml
- pack: Vendor # all packs of the "Vendor" are considered, latest version only
- pack: Vendor::Nam* # all packs of the "Vendor" that match "Nam*" wildcard are considered, latest version only
```

And for unspecified `packs:`, we take into account the entire pack repository, but only consider latest pack versions.

**Comment by ReinhardKeil** [comment]
@brondani what would be the effort to change behavior to load just the latest version of each pack?

**Comment by brondani** [comment]
> what would be the effort to change behavior to load just the latest version of each pack?
The required effort should be reasonably low.

**Comment by mcgordonite** [comment]
Is this something we can do for Csolution 2.0?

**Comment by jkrech** [comment]
Adding proposal 2 to the scope for 2.0.0.

**Comment by jkrech** [comment]
Available starting CMSIS-Toolbox 2.0.0-dev1
